trk/src/NfcUtils/writeClimb.js

```diff
@@ -19,6 +32,61 @@ async function writeClimb(climbID) {
     allBytes = [...allBytes, ...block];
   }
 
+  //Now adding the grade right after
+  //Grade will occupy exactly 2 blocks
+  // Totally blocks 4-10 (6 blocks) store data UPTO NOW, 15 onwards is the signature
+  let block = [];
+  let j = 0;
+  for (; j < 4; j++) {
+    if (j<grade.length) {
+      block.push(grade.charCodeAt(j));
+    }
+    else {
+      block.push(32);
+    }
+  }
+  blocks.push(block);
+  allBytes = [...allBytes, ...block];
+  block = [];
+  while (j < 8) {
+    if (j<grade.length) {
+      block.push(grade.charCodeAt(j));
+    }
+    else {
+      block.push(32);
+    }
+    j=j+1;
+  }
+  blocks.push(block);
+  //not adding it to returned bytes 
+  allBytes = [...allBytes, ...block];
+
+  //Now adding name (3 blocks)
+  for (let i = 0; i < 12; i += 4) {
+    if (i+4 <= name.length) {
+      let subStr = name.substring(i, i + 4);
+      let block = [];
+      for (let j = 0; j < subStr.length; j++) {
+        block.push(subStr.charCodeAt(j));
+      }
+      blocks.push(block);
+      allBytes = [...allBytes, ...block];
+    }
+    else {
+      let subStr = name.substring(i);
+      let block = [];
+      for (let j = 0; j < 4; j++) {
+        if (j<subStr.length) {
+          block.push(subStr.charCodeAt(j));
+        }
+        else {
+          block.push(32);
+        }
+      }
+      blocks.push(block);
+      allBytes = [...allBytes, ...block];
+    }
+  }
```

Comment:
This is great. Well done. A little more documentation would be great, but appreciate the effort of leaving comments at the top. Again, good job

Reply:
Will do from now on! Just repeated the same process......